File: C/练习/code_13_1.c
#include <stdio.h>


// 打印水仙花数

void print_num(void) {
    int x, y, z;
    for (x = 1; x <= 9; x++) {
        for (y = 0; y <= 9; y++) {
            for (z = 0; z <= 9; z++) {
                if (x * x * x + y * y * y + z * z * z == x * 100 + y * 10 + z) {
                    printf("%d%d%d 是水仙花数\n", x, y, z);
                }
            }
        }
    }
}



int main(void) {
    print_num();
    return 0;
}

File: C/Replicate_With_Python/code_13_1.py
def print_num():
    for x in range(1, 10):
        for y in range(0, 10):
            for z in range(0, 10):
                if x ** 3 + y ** 3 + z ** 3 == x * 100 + y * 10 + z:
                    print(f"{x}{y}{z} 是水仙花数")
    return


def main():
    print_num()


if __name__ == '__main__':
    main()
